dbos/dbos_test.go

```diff
@@ -58,6 +61,58 @@ func TestConfig(t *testing.T) {
 		assert.Equal(t, expectedMsg, dbosErr.Message)
 	})
 
+	t.Run("NewSystemDatabaseWithCustomPool", func(t *testing.T) {
+
+		// Logger
+		var buf bytes.Buffer
+		slogLogger := slog.New(slog.NewTextHandler(&buf, &slog.HandlerOptions{
+			Level: slog.LevelDebug,
+		}))
+
+		slogLogger = slogLogger.With("service", "dbos-test", "environment", "test")
+
+		// Custom Pool
+		poolConfig, err := pgxpool.ParseConfig(databaseURL)
+		require.NoError(t, err)
+
+		poolConfig.MaxConns = 10
+		poolConfig.MinConns = 5
+		poolConfig.MaxConnLifetime = 2 * time.Hour
+		poolConfig.MaxConnIdleTime = time.Minute * 2
+
+		poolConfig.ConnConfig.ConnectTimeout = 10 * time.Second
+
+		pool, err := pgxpool.NewWithConfig(context.Background(), poolConfig)
+		require.NoError(t, err)
+
+		config := Config{
+			DatabaseURL:  databaseURL,
+			AppName:      "test-custom-pool",
+			Logger:       slogLogger,
+			SystemDBPool: pool,
+		}
+
+		customdbosContext, err := NewDBOSContext(context.Background(), config)
+		require.NoError(t, err)
+		require.NotNil(t, customdbosContext)
+
+		dbosCtx, ok := customdbosContext.(*dbosContext)
+		require.True(t, ok)
```

Comment:
You need to defer the shutdown of the DBOS context, that's why the tests are failing

```go
defer dbosCtx.Shutdown(10 * time.Second)
```